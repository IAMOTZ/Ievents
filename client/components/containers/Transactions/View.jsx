import React from 'react';
import PropTypes from 'prop-types';
import SideNavigation from '../../common/SideNavigation';
import Header from '../../common/Header';
import EventCards from '../../common/EventCards';
import { AuthTopNavigation } from '../../common/TopNavigation';
import { ConfirmationModal, EventDetailsModal } from '../../common/Modals';
import { LoadingBox } from '../../common/LoadingAnimation';
import Footer from '../../common/Footer';
import Pagination from '../../common/Pagination';

const View = props => (
  <div id="events-container">
    <AuthTopNavigation
      name={props.userName}
      title="Transaction/Events"
      isAdmin={props.isAdmin}
      isSuperAdmin={props.isSuperAdmin}
      dispatch={props.dispatch}
    />
    <div className="container-fluid">
      <div className="row">
        <SideNavigation
          name={props.userName}
          isAdmin={props.isAdmin}
          isSuperAdmin={props.isSuperAdmin}
          dispatch={props.dispatch}
        />
        <div className="col-lg-10 offset-md-2 mt-lg-0">
          <Header text="Transaction/Events" />
          <div className="page-content">
            {
              props.transactions.length === 0 && props.fetchingTransactionsStarted ?
                <LoadingBox iconSize={4} /> :
                <div className="row">
                  <EventCards
                    events={props.transactions}
                    startEventCancel={props.startEventCancel}
                    stopEventCancel={props.stopEventCancel}
                    finishEventCancel={props.finishEventCancel}
                    deletingTransactionStarted={props.deletingTransactionStarted}
                    createModalContent={props.createModalContent}
                    modalContent={props.modalContent}
                    isTransactionsPage
                  />
                </div>
            }
            <Pagination
              pageCount={Math.ceil(props.pagination.totalCount / props.pagination.limit)}
              onPageChange={props.updatePagination}
            />
            <ConfirmationModal
              onCancel={props.stopEventCancel}
              onOK={props.finishEventCancel}
            ><span>Are you sure you want to cancel this event?</span>
            </ConfirmationModal>
            <EventDetailsModal {...props.modalContent} />
          </div>

        </div>
      </div>
    </div>
    <span className="d-block d-sm-none mt-5">
      <Footer />
    </span>
  </div>
);

/* eslint-disable react/forbid-prop-types */
View.propTypes = {
  userName: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  isSuperAdmin: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
  transactions: PropTypes.array,
  startEventCancel: PropTypes.func.isRequired,
  fetchingTransactionsStarted: PropTypes.bool.isRequired,
  deletingTransactionStarted: PropTypes.bool.isRequired,
  finishEventCancel: PropTypes.func.isRequired,
  stopEventCancel: PropTypes.func.isRequired,
  createModalContent: PropTypes.func.isRequired,
  modalContent: PropTypes.object,
  pagination: PropTypes.object,
  updatePagination: PropTypes.func.isRequired,
};

View.defaultProps = {
  modalContent: null,
  transactions: [],
  pagination: {},
};

export default View;
