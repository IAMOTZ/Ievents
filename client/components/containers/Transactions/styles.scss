#transactions-container {
  #transactions {
    margin-top: 100px;
    .card {
      z-index: 4;
      a:hover {
        text-decoration: none;      
      }
      .card-header {
        background-color: white;
      }
    }
  }
}