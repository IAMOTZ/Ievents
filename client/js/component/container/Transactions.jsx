import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import {
  getAllTransactions,
  allowTransaction,
  cancelTransaction,
  deleteTransaction,
  clearStatus
} from '../../actions/transactionActions';

import UserSideNav from '../common/SideNavigation.jsx';
import TransactionCards from '../common/TransactionCards.jsx';
import { UserTopNav } from '../common/TopNavigation.jsx';
import { WarningAlert } from '../common/Alert';

@connect((store) => {
  return {
    user: store.user.user,
    authenticated: store.user.status.fetched,
    transactions: store.transactions,
  }
})

export default class Transactions extends React.Component {
  componentWillMount() {
    this.props.dispatch(getAllTransactions(this.props.user.token));
  }

  refresh = () => {
    this.props.dispatch(clearStatus('ALL'));
    this.props.dispatch(getAllTransactions(this.props.user.token));
  }

  componentDidUpdate() {
    if (this.props.transactions.status.allowed ||
      this.props.transactions.status.canceled ||
      this.props.transactions.status.deleted) {
      this.refresh();
    }
  }

  cancelTransaction = (e) => {
    this.props.dispatch(cancelTransaction(this.props.user.token, e.target.dataset.transactionId));
  }

  allowTransaction = (e) => {
    this.props.dispatch(allowTransaction(this.props.user.token, e.target.dataset.transactionId));
  }

  deleteTransaction = (e) => {
    this.props.dispatch(deleteTransaction(this.props.user.token, e.target.dataset.transactionId));
  }

  render() {
    if (!this.props.authenticated) {
      return (<Redirect to="/users/login" />)
    } else {
      return (
        <div id="transactions-container">
          {/* Top navigation on small screen */}
          <UserTopNav name={this.props.user.name} title='Transactions' />

          <div class="container-fluid">
            <div class="row">

              {/*  Side navigation on large screen */}
              <UserSideNav userName={this.props.user.name} />

              {/* Main content */}
              <div class="col-lg-10 offset-md-2" id="add-event-section">

                {/* Content Header(navigation) on large screen */}
                <nav className="navbar w-100 mt-3 d-none d-lg-flex justify-content-between">
                  <a className="navbar-brand text-white">
                    <strong>Transactions</strong>
                  </a>
                  <button className="btn btn-primary" onClick={this.refresh}>Refresh</button>
                </nav>

                <div id="transactions" class="mt-lg-0">
                  <div id="accordion" role="tablist">
                    <div className="text-center d-lg-none">
                      <button className="btn btn-primary" onClick={this.refresh}>Refresh</button>
                    </div>
                    <TransactionCards
                      centers={this.props.transactions.centers}
                      onCancel={this.cancelTransaction}
                      onAllow={this.allowTransaction}
                      onDelete={this.deleteTransaction} />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      )
    }
  }
}