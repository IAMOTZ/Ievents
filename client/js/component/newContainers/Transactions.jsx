import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import {
  getAllTransactions, deleteTransaction,
} from '../../actions/transactionActions';
import { stopAsyncProcess } from '../../actions/commonActions';
import * as asyncProcess from '../../actions/asyncProcess';
import UserSideNav from '../common/SideNavigation';
import TransactionCards from '../common/TransactionCards';
import { ConfirmModal } from '../common/Modal';
import { UserTopNav } from '../common/TopNavigation';
import { LoadingContainer, LoadingIcon } from '../common/LoadingAnimation';

@connect((store) => {
  const { user } = store.authReducer;
  return {
    user,
    isAdmin: user.role === 'admin',
    isSuperAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isUserAuthenticaed: store.authReducer.loggingUserResolved,
    transactions: store.fetchTransactionsReducer.transactions,
    fetchingTransactionsStarted: store.fetchTransactionsReducer.fetchingTransactionsStarted,
    deletingTransactionStarted: store.deleteTransactionReducer.deletingTransactionStarted,
    deletingTransactionResolved: store.deleteTransactionReducer.deletingTransactionResolved,
  };
})
class Transactions extends React.Component {
  constructor() {
    super();
    this.state = {
      toDelete: null,
      confirmModalVisible: false,
    };
  }

  componentWillMount() {
    this.props.dispatch(getAllTransactions(this.props.user.token));
  }

  componentDidUpdate() {
    if (this.props.deletingTransactionResolved) {
      this.setState({ toDelete: null });
      this.refresh();
    }
  }

  /**
   * Refresh the page by clearing all created info eg Error Messages.
   */
  refresh = () => {
    this.props.dispatch(stopAsyncProcess(asyncProcess.DELETING_TRANSACTION));
    this.props.dispatch(getAllTransactions(this.props.user.token));
  }

  /**
   * It updates the state about a transaction that is to be deleted/canceled.
   * It also initiates the confirmation modal.
   * @param {Event} event The event object.
   */
  startDelete = (event) => {
    this.setState({
      toDelete: event.target.dataset.transactionId,
      confirmModalVisible: !this.state.confirmModalVisible,
    });
  }

  /**
   * It eventually deletes the event and hides back the modal.
   */
  finishDelete = () => {
    this.props.dispatch(deleteTransaction(this.props.user.token, this.state.toDelete));
    this.setState({ confirmModalVisible: !this.state.confirmModalVisible });
  }

  /**
   * It cancels the deleting and also hides back the modal.
   */
  cancelDelete = () => {
    this.setState({
      toDelete: null,
      confirmModalVisible: !this.state.confirmModalVisible,
    });
  }

  render() {
    if (!this.props.isUserAuthenticaed) {
      return (<Redirect to="/users/login" />);
    }
    return (
      <div id="transactions-container">
        {/* Top navigation on small screen */}
        <UserTopNav
          name={this.props.user.name}
          title="Transactions"
          isAdmin={this.props.isAdmin}
          isSuperAdmin={this.props.isSuperAdmin}
          dispatch={this.props.dispatch}
        />
        <div className="container-fluid">
          <div className="row">
            {/*  Side navigation on large screen */}
            <UserSideNav
              name={this.props.user.name}
              isAdmin={this.props.isAdmin}
              isSuperAdmin={this.props.isSuperAdmin}
              dispatch={this.props.dispatch}
            />
            {/* Main content */}
            <div className="col-lg-10 offset-md-2" id="add-event-section">
              {/* Content Header(navigation) on large screen */}
              <nav className="navbar w-100 mt-3 d-none d-lg-flex justify-content-between">
                <a className="navbar-brand text-white">
                  <strong>Transactions</strong>
                </a>
                <button
                  className="btn btn-primary pointer-button"
                  disabled={this.props.fetchingTransactionsStarted}
                  onClick={this.refresh}
                >Refresh
                </button>
              </nav>
              <div id="transactions" className="mt-lg-0">
                <div id="accordion" role="tablist">
                  {
                    this.props.transactions.length === 0 &&
                      this.props.fetchingTransactionsStarted ?
                        <LoadingContainer iconSize={4} /> :
                        <div>
                          <div className="text-center d-lg-none">
                            <button
                              className="btn btn-primary"
                              disabled={this.props.fetchingTransactionsStarted}
                              onClick={this.refresh}
                            >Refresh
                            </button>
                          </div>
                          <LoadingIcon start={this.props.fetchingTransactionsStarted} size={2} />
                          <TransactionCards
                            centers={this.props.transactions}
                            onCancel={this.startDelete}
                            toDelete={this.state.toDelete}
                            deleting={this.props.deletingTransactionStarted}
                          />
                        </div>
                  }
                  <ConfirmModal
                    visible={this.state.confirmModalVisible}
                    onCancel={this.cancelDelete}
                    onOK={this.finishDelete}
                  >
                    <span>Are you sure you want to cancel this event?</span>
                  </ConfirmModal>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Transactions;
