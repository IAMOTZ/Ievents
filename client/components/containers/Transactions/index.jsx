/* global $ */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  getAllTransactions, deleteTransaction,
} from '../../../actions/transactionActions';
import { stopAsyncProcess } from '../../../actions/commonActions';
import * as asyncProcess from '../../../actions/asyncProcess';
import './styles.scss';
import View from './View';

@connect((store) => {
  const { user } = store.authReducer;
  return {
    userName: user.name,
    userToken: user.token,
    isAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isSuperAdmin: user.role === 'superAdmin',
    transactions: store.fetchTransactionsReducer.transactions,
    fetchingTransactionsStarted: store.fetchTransactionsReducer.fetchingTransactionsStarted,
    deletingTransactionStarted: store.deleteTransactionReducer.deletingTransactionStarted,
    deletingTransactionResolved: store.deleteTransactionReducer.deletingTransactionResolved,
  };
})
class Transactions extends React.Component {
  constructor() {
    super();
    this.state = {
      toDelete: null,
    };
  }

  componentWillMount() {
    this.props.dispatch(getAllTransactions(this.props.userToken));
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.deletingTransactionResolved) {
      this.setState({ toDelete: null });
      this.refresh();
    }
  }

  /**
   * Refresh the page by clearing all created info eg Error Messages.
   */
  refresh = () => {
    this.props.dispatch(stopAsyncProcess(asyncProcess.DELETING_TRANSACTION));
    this.props.dispatch(getAllTransactions(this.props.userToken));
  }

  /**
   * It updates the state about a transaction that is to be deleted/canceled.
   * @param {Event} event The event object.
   */
  startDelete = (event) => {
    this.setState({
      toDelete: event.target.dataset.transactionId,
    });
  }

  /**
   * It eventually deletes the event.
   */
  finishDelete = () => {
    this.props.dispatch(deleteTransaction(this.props.userToken, this.state.toDelete));
    $('#confirmation-modal').modal('hide');
  }

  /**
   * It cancels the deleting of a transaction.
   */
  cancelDelete = () => {
    this.setState({ toDelete: null });
  }

  render() {
    return (
      <View
        userName={this.props.userName}
        isAdmin={this.props.isAdmin}
        isSuperAdmin={this.props.isSuperAdmin}
        dispatch={this.props.dispatch}
        fetchingTransactionsStarted={this.props.fetchingTransactionsStarted}
        deletingTransactionStarted={this.props.deletingTransactionStarted}
        transactions={this.props.transactions}
        refresh={this.refresh}
        toDelete={this.state.toDelete}
        startDelete={this.startDelete}
        cancelDelete={this.cancelDelete}
        finishDelete={this.finishDelete}
      />
    );
  }
}

Transactions.defaultProps = {
  userName: '',
  userToken: '',
  isAdmin: false,
  isSuperAdmin: false,
  transactions: [],
  fetchingTransactionsStarted: false,
  deletingTransactionStarted: false,
  deletingTransactionResolved: false,
  dispatch: () => {},
};

/* eslint-disable react/forbid-prop-types */
Transactions.propTypes = {
  userName: PropTypes.string,
  userToken: PropTypes.string,
  isAdmin: PropTypes.bool,
  isSuperAdmin: PropTypes.bool,
  transactions: PropTypes.array,
  fetchingTransactionsStarted: PropTypes.bool,
  deletingTransactionStarted: PropTypes.bool,
  deletingTransactionResolved: PropTypes.bool,
  dispatch: PropTypes.func,
};

export default Transactions;
