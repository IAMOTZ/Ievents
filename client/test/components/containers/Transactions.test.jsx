import React from 'react';
import Transactions from '../../../components/containers/Transactions';
import TransactionsView from '../../../components/containers/Transactions/View';

const sampleTransaction = {
  id: 1,
  centerId: 1,
  userId: 1,
  title: 'test-event',
  description: 'test description',
  date: '12/34/2020',
  status: 'allowed',
  user: {
    email: 'tester@gmail.com'
  }
};

describe('<Transactions />', () => {
  const props = {
    userName: 'tester',
    userToken: 'sampleToken',
    isAdmin: false,
    isSuperAdmin: false,
    centerToTransact: 3,
    transactions: [sampleTransaction],
    pagination: {},
    fetchingTransactionsStarted: false,
    cancelingTransactionStarted: false,
    cancelingTransactionResolved: false,
    dispatch: jest.fn(),
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Transactions {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should redirect to the centers-as-transactions page if there is no center to transact', () => {
      wrapper.setProps(alterProps({ centerToTransact: null }));
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behaviour:', () => {
    let wrapper; let wrapperInstance;
    beforeEach(() => {
      props.dispatch.mockReset();
      wrapper = shallow(<Transactions {...props} />);
      wrapperInstance = wrapper.instance();
    });
    // The names on the following test block refers
    // to the instance method that handles the behaviour tested.
    describe('COMOPONENT_WILL_MOUNT', () => {
      it('should dispatch an action', () => {
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
    describe('COMOPONENT_WILL_RECIEVE_PROPS', () => {
      it('should dispatch two actions and remove the center to cancel from the state if canceling transaciton resolved', () => {
        props.dispatch.mockReset();
        wrapper.setProps(alterProps({ cancelingTransactionResolved: true }));
        expect(wrapper.state('toCancel')).toBeNull();
        expect(props.dispatch.mock.calls.length).toEqual(2);
      });
    });
    describe('REFRESH', () => {
      it('should dispatch two actions', () => {
        props.dispatch.mockReset();
        wrapperInstance.refresh();
        expect(props.dispatch.mock.calls.length).toEqual(2);
      });
    });

    describe('START_EVENT_CANCEL', () => {
      it('should update the state with id of event to cancel', () => {
        const fakeEvent = { currentTarget: { id: sampleTransaction.id } };
        wrapperInstance.startEventCancel(fakeEvent);
        expect(wrapper.state('toCancel')).toEqual(sampleTransaction.id);
      });
    });

    describe('REFRESH', () => {
      it('should dispatch an actions', () => {
        props.dispatch.mockReset();
        wrapperInstance.finishEventCancel();
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });

    describe('STOP_EVENT_CANCEL', () => {
      it('should remove the id of event to cancel from the state', () => {
        wrapperInstance.stopEventCancel();
        expect(wrapper.state('toCancel')).toBeNull();
      });
    });

    describe('CREATE_MODAL_CONTENT', () => {
      it('should update the state with the content of a modal', () => {
        wrapperInstance.createModalContent(sampleTransaction);
        expect(wrapper.state('modalContent')).toEqual({
          eventTitle: sampleTransaction.title,
          eventDescription: sampleTransaction.description,
        });
      });
    });

    describe('UPDATE_PAGINATION', () => {
      it('should dispatch an action', () => {
        props.dispatch.mockReset();
        const fakePageData = { selected: 3 };
        wrapperInstance.updatePagination(fakePageData);
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
  });
});

describe('<TransactionsView />', () => {
  const props = {
    userName: 'tester',
    isAdmin: false,
    isSuperAdmin: false,
    dispatch: jest.fn(),
    transactions: [],
    startEventCancel: () => {},
    fetchingTransactionsStarted: false,
    cancelingTransactionStarted: false,
    finishEventCancel: () => {},
    stopEventCancel: () => {},
    createModalContent: () => {},
    modalContent: {},
    pagination: {},
    updatePagination: () => {},
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<TransactionsView {...props} />);
    });
    it('should render correctly when there are no transactions and it is fetching', () => {
      wrapper.setProps(alterProps({ fetchingTransactionsStarted: true }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly when there are transactions and it is not fetching', () => {
      wrapper.setProps(alterProps({ events: [sampleTransaction] }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
