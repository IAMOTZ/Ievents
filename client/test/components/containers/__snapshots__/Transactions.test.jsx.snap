// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transactions /> Rendering: should redirect to the centers-as-transactions page if there is no center to transact 1`] = `
<Redirect
  push={false}
  to="/centers/transactions"
/>
`;

exports[`<Transactions /> Rendering: should render correctly 1`] = `
<View
  cancelingTransactionStarted={false}
  createModalContent={[Function]}
  dispatch={
    [MockFunction] {
      "calls": Array [
        Array [
          [Function],
        ],
      ],
    }
  }
  fetchingTransactionsStarted={false}
  finishEventCancel={[Function]}
  isAdmin={false}
  isSuperAdmin={false}
  isTransactionsPage={true}
  modalContent={null}
  pagination={Object {}}
  refresh={[Function]}
  startEventCancel={[Function]}
  stopEventCancel={[Function]}
  toCancel={null}
  transactions={
    Array [
      Object {
        "centerId": 1,
        "date": "12/34/2020",
        "description": "test description",
        "id": 1,
        "status": "allowed",
        "title": "test-event",
        "user": Object {
          "email": "tester@gmail.com",
        },
        "userId": 1,
      },
    ]
  }
  updatePagination={[Function]}
  userName="tester"
/>
`;

exports[`<TransactionsView /> Rendering should render correctly when there are no transactions and it is fetching 1`] = `
<div
  id="events-container"
>
  <AuthTopNavigation
    dispatch={[MockFunction]}
    isAdmin={false}
    isSuperAdmin={false}
    name="tester"
    title="Transaction/Events"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <SideNavigation
        dispatch={[MockFunction]}
        isAdmin={false}
        isSuperAdmin={false}
        name="tester"
      />
      <div
        className="col-lg-10 offset-md-2 mt-lg-0"
      >
        <Header
          text="Transaction/Events"
        />
        <div
          className="page-content"
        >
          <LoadingBox
            iconSize={4}
          />
          <Pagination
            onPageChange={[Function]}
            pageCount={NaN}
          />
          <ConfirmationModal
            onCancel={[Function]}
            onOK={[Function]}
          >
            <span>
              Are you sure you want to cancel this event?
            </span>
          </ConfirmationModal>
          <EventDetailsModal
            eventDescription={null}
            eventTitle={null}
          />
        </div>
      </div>
    </div>
  </div>
  <span
    className="d-block d-sm-none mt-5"
  >
    <Footer />
  </span>
</div>
`;

exports[`<TransactionsView /> Rendering should render correctly when there are transactions and it is not fetching 1`] = `
<div
  id="events-container"
>
  <AuthTopNavigation
    dispatch={[MockFunction]}
    isAdmin={false}
    isSuperAdmin={false}
    name="tester"
    title="Transaction/Events"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <SideNavigation
        dispatch={[MockFunction]}
        isAdmin={false}
        isSuperAdmin={false}
        name="tester"
      />
      <div
        className="col-lg-10 offset-md-2 mt-lg-0"
      >
        <Header
          text="Transaction/Events"
        />
        <div
          className="page-content"
        >
          <div
            className="page-content text-center"
          >
            <h1
              className="display-3"
            >
              The center has no transactions
            </h1>
            <h3
              className="font-weight-normal"
            >
              Back toÂ 
              <Link
                replace={false}
                to="/centers/transactions"
              >
                transactions page
              </Link>
              .
            </h3>
          </div>
          <Pagination
            onPageChange={[Function]}
            pageCount={NaN}
          />
          <ConfirmationModal
            onCancel={[Function]}
            onOK={[Function]}
          >
            <span>
              Are you sure you want to cancel this event?
            </span>
          </ConfirmationModal>
          <EventDetailsModal
            eventDescription={null}
            eventTitle={null}
          />
        </div>
      </div>
    </div>
  </div>
  <span
    className="d-block d-sm-none mt-5"
  >
    <Footer />
  </span>
</div>
`;
